app/models/post.rb

```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
-  has_many :likes_counter, foreign_key: 'post_id'
-  has_many :comment_counter, foreign_key: 'post_id'
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [OPTIONAL] you can use the one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` on the `update_posts_counter` or any `update_resource_counter` method and make the method private 💡

Reply:
Thanks for this suggestion. I have included it.